src/avif.c

```diff
@@ -667,7 +667,9 @@ avifBool avifCropRectConvertCleanApertureBox(avifCropRect * cropRect,
                                              avifPixelFormat yuvFormat,
                                              avifDiagnostics * diag)
 {
-    avifDiagnosticsClearError(diag);
+    if (diag) {
+        avifDiagnosticsClearError(diag);
+    }
```

Comment:
Could you explain why this is needed?

If you need to call these two functions from places where you do not have a `avifDiagnostics` object, we have been working around this by declaring a local `avifDiagnostics` variable and passing it to the function.

Reply:
Looking at this closer, I found that `avifDiagnosticsPrintf()` already ignores a null `diag` input. For API consistency it seems better to have `avifDiagnosticsClearError()` ignore a null `diag` input.

Reply:
> Could you explain why this is needed?
> 
> If you need to call these two functions from places where you do not have a `avifDiagnostics` object, we have been working around this by declaring a local `avifDiagnostics` variable and passing it to the function.

Yes this is to avoid adding a line in the unrelated `avifconitest.cc` from #1432.

> Looking at this closer, I found that `avifDiagnosticsPrintf()` already ignores a null `diag` input. For API consistency it seems better to have `avifDiagnosticsClearError()` ignore a null `diag` input.

Done.

Reply:
Thank you for the reply. I looked into the history of the relevant code. I found no evidence of passing a null `diag` to `avifDiagnosticsPrintf()`. Also, since we always call `avifDiagnosticsClearError()` first and it disallows a null `diag`, I think no code can possibly pass a null `diag` to `avifDiagnosticsPrintf()`. So it seems better to remove the null `diag` check from `avifDiagnosticsPrintf()`, and then just pass a local `diag` variable in `avifconitest.cc` (and use it in an error message):

```
    avifDiagnostics diag;
    ASSERT_TRUE(avifCleanApertureBoxConvertCropRect(
        &image->clap, &rect, image->width, image->height, image->yuvFormat,
        &diag)) << diag.error;
```

This way we can detect the bug of passing a null `diag` by mistake.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1496

Reply:
Thanks a lot for tracking down the code that passes a null `diag`. With your hints, I also found https://github.com/AOMediaCodec/libavif/commit/388889c2. Given the new information, I suggest we go ahead and merge this pull request.